runtime/metricsview/executor_summary.go

```
@@ -0,0 +1,250 @@
+		if !e.security.CanAccessField(dim.Name) {
+			continue
+		}
+
+		// If we have reached the limit, stop adding dimensions
+		if len(dimensions)+len(timeDimensions) >= DefaultSummaryLimit {
+			break
+		}
+
+		// If the dimension is the time dimension, handle it separately
+		if dim.Name == timeDimension {
+			timeDimensions = append(timeDimensions, dim)
+		} else {
+			dimensions = append(dimensions, dim)
+		}
+	}
+
+	// Get the default time range for the metrics view
+	defaultTimeRange, err := e.Timestamps(ctx, timeDimension)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get default time range: %w", err)
+	}
+
+	timeDimensionSummaries := make([]DimensionSummary, 0, len(timeDimensions))
+
+	// Validate time range
+	if defaultTimeRange.Max.IsZero() {
+		return &SummaryResult{
+			Dimensions: timeDimensionSummaries,
+			TimeRange:  defaultTimeRange,
+		}, nil
+	}
+
+	for _, dim := range timeDimensions {
+		timeRange, err := e.Timestamps(ctx, dim.Name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get time range for dimension %s: %w", dim.Name, err)
+		}
+
+		var dataType string
+		if dim.DataType != nil {
+			dataType = dim.DataType.Code.String()
+		}
+
+		timeDimensionSummaries = append(timeDimensionSummaries, DimensionSummary{
+			Name:      dim.Name,
+			TimeRange: timeRange,
+			DataType:  dataType,
+		})
+	}
+
+	// Create summaries for normal dimensions
+	if len(dimensions) == 0 {
+		return &SummaryResult{
+			Dimensions: timeDimensionSummaries,
+			TimeRange:  defaultTimeRange,
+		}, nil
+	}
+
+	// Build and execute the summary query
+	selectClauses := make([]string, 0, len(dimensions)*4)
+	var failedDimensions []string
+
+	for _, dim := range dimensions {
+		dimName := dim.Name
+		expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+		if err != nil {
+			failedDimensions = append(failedDimensions, dimName)
+			continue // Skip this dimension instead of failing entirely
+		}
```

You can return the error here immediately. It would be an internal error because this stuff is already validated upstream (`e.metricsView` comes from `MetricsView.State.ValidSpec`)